### PR TITLE
More serialization preparation 

### PR DESCRIPTION
**Pull request by akva2**

### CODE REVIEW

**Comment by akva2** [comment]
jenkins build this opm-simulators=2257 please

**Comment by akva2** [comment]
jenkins build this opm-simulators=2257 please

File: opm/parser/eclipse/EclipseState/Schedule/Well/Well.hpp [hunk 1]
```diff
@@ -383,6 +384,35 @@ class Well {
...
+         Status status,
+         double drainageRadius,
+         bool allowCrossFlow,
+         bool automaticShutIn,
+         bool isProducer,
+         const WellGuideRate& guideRate,
+         double efficiencyFactor,
+         double solventFraction,
+         bool prediction_mode,
+         std::shared_ptr<const WellEconProductionLimits> econLimits,
```
Comment (joakim-hove) [inline opm/parser/eclipse/EclipseState/Schedule/Well/Well.hpp]: If possible I would prefer to let the use of `std::shared_ptr<>` be an implementation detail - not exposed in the API.

Comment (akva2) [reply opm/parser/eclipse/EclipseState/Schedule/Well/Well.hpp]: well, i'm not really using shared_ptr to reflect internal semantics of the Well class here. I need to move a pointer between instances and using a shared_ptr is a convenient and safe way to do so.

Comment (joakim-hove) [reply opm/parser/eclipse/EclipseState/Schedule/Well/Well.hpp]: OK - thanks for the clarification.

**Comment by akva2** [comment]
review/merge?

**Comment by joakim-hove** [comment]
> review/merge?

I must admit these changes rub me a bit the wrong way - but I understand the need.

**Pull request merged**